rule.php

```diff
@@ -22,9 +22,18 @@
  * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
-defined('MOODLE_INTERNAL') || die();
+defined ( 'MOODLE_INTERNAL' ) || die ();
 
-require_once($CFG->dirroot . '/mod/quiz/accessrule/accessrulebase.php');
+global $CFG;
+
+// This work-around is required until Moodle 4.2 is the lowest version we support.
+if (class_exists('\mod_quiz\local\access_rule_base')) {
```

Comment:
It's better to use version comparision
```
if ($CFG->version >= 2023042400.00) { // Moodle 4.2.
```

Reply:
Are you sure, @vadimonus ?
Someone could have locally backported the 4.2 class.
I do prefer 
`if (!(class_exists('\mod_quiz\local\access_rule_base'))) { `
(given that you want to turn the logic around).